### PR TITLE
Sites Dashboard: Skeleton for new dashboard

### LINKED ISSUE #65163: Sites Management Dashboard: create a skeleton for the new interface
Let's create a skeleton for the new UI:

Screenshot 2022-07-01 at 22 17 47

It should contain the following:

- Header Section
  - My Sites title,
  - `New Site` button linking to `wordpress.com/start`.
- Main Section
 - A container with dummy data that will hold the future page content (tab view will be introduced in a separate task).

Done is:
- [x] The UI components match the proposed Figma designs.
- [x] It works as designed for smaller screen sizes.

### PR DESCRIPTION
**Pull request by p-jackson**
#### Proposed Changes

CleanShot 2022-07-04 at 18 27 18@2x

CleanShot 2022-07-04 at 18 28 44@2x

This addresses the first "create a skeleton" task in #65163. It does a bit more than the acceptance criteria, but I figured it was worth committing the table even though it's not finished.

It's a biggish commit, but I've tried to split the commits into more readable chunks if that helps.

The parts that are important for the acceptance criteria:
- Header elements have the correct spacing
- Mobile layout works
- Heading typography is correct
- Links to the signup flow
- Anything else you see mentioned in https://github.com/Automattic/wp-calypso/issues/65163

Some issues that I don't think block this PR:
- [I'm currently importing Calypso code into the components package](https://github.com/Automattic/wp-calypso/pull/65182/files#diff-fc32c0effaafe455f86a656313c03beb411c2996b04604fe9ff18f976074ffb9R3). Discussing refactoring things so that's not necessary. p1656917133483219/1656482582.511659-slack-C02DQP0FP
- After clicking "New Sites" the back button in the top-left of the site creation flow doesn't return the user back to the dashboard. I've created a task for that https://github.com/Automattic/wp-calypso/issues/65183

#### Testing Instructions

* Head to `calypso.localhost:3000/sites-dashboard` to try it out
* Use the "New site" button
* Try a phone layout

Fixes #65163

### CODE REVIEW

**Comment by p-jackson** [comment]
@SaxonF component consistency check: buttons in Calypso currently use sentence caps. Should we do the same here and use "New site" instead of "New Site"?

File: packages/components/src/sites-table/sites-table-filter-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		>
+			{ ( tab ) => renderContents( filterSites( allSites, tab.name ) ) }
+		</TabPanel>
+	);
+}
+
+function filterSites( sites: SiteData[], filterType: string ): SiteData[] {
+	return sites.filter( ( site ) => {
+		const isComingSoon =
+			site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' );
```
Comment (sejas) [inline packages/components/src/sites-table/sites-table-filter-tabs.tsx]: Something here is not loading my "coming soon" sites.
I'm fixing it in my PR where I add the tabs. https://github.com/Automattic/wp-calypso/pull/65206

UPDATE:
It's the line 44.

```diff
			case 'coming-soon':
-				return site.is_private && isComingSoon;
+				return isComingSoon;
```

**Review by sejas — APPROVED** [review]
@p-jackson Wow!! This looks super cool!
Thanks for creating this "skeleton". I love that you are using styled components.
I've seen you have also implemented the tabs and you are loading real data. ❤️ 

So much good work so fast and in so few lines 💪 .

File: packages/components/src/sites-table/sites-table-filter-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	children: renderContents,
+	className,
+}: SitesTableFilterTabsProps ) {
+	const { __ } = useI18n();
+
+	return (
+		<TabPanel
+			className={ className }
+			tabs={ [
+				{ name: 'all', title: __( 'All', __i18n_text_domain__ ) },
```
Comment (zaguiini) [inline packages/components/src/sites-table/sites-table-filter-tabs.tsx]: Will we need to add the `__i18n_text_domain__` for _all_ instances of `__()`? If yes, how about we create a proxy hook that injects this second argument for us?

**Comment by vykes-mac** [comment]
Nice work @p-jackson , this skeleton has a good amount of flesh we can build on. Re styled components I've read somewhere that 

> If your application is expected to have frequent updates that might affect performance, consider switching from runtime styling solutions ([Chakra UI](https://chakra-ui.com/), [emotion](https://emotion.sh/docs/introduction), [styled-components](https://styled-components.com/) that generate styles during runtime) to zero runtime styling solutions ([tailwind](https://tailwindcss.com/), [linaria](https://github.com/callstack/linaria), [vanilla-extract](https://github.com/seek-oss/vanilla-extract), [CSS modules](https://github.com/css-modules/css-modules) which generate styles during build time).

It might not necessarily apply here but putting it here for information purposes.

File: packages/components/src/sites-table/sites-table-filter-tabs.tsx (hunk 2 above)
Comment (p-jackson) [reply packages/components/src/sites-table/sites-table-filter-tabs.tsx]: It's only necessary for instances that are inside the `packages/` folder. Anything inside `client/sites-dashboard/` folder don't need it.

Unfortunately it's not possible to write a wrapper. It's the same reason you can't pass a string variable to the `__()` function: our translation tools statically analyse the code looking for those `__` functions, so they have to be in the form that the tool is able to understand.
I suppose it is possible that we modify our translation tools to be more aware of packages. You can read more about possible solutions here: pbAok1-1Ao-p2

File: packages/components/src/sites-table/sites-table-filter-tabs.tsx (hunk 1 above)
Comment (p-jackson) [reply packages/components/src/sites-table/sites-table-filter-tabs.tsx]: Yeah, this logic is a bit of a mess 😅 I tried to copy the logic that shows the `[Coming Soon]` and `[Private]` badges in the site switcher but I think I messed it up. I figured it was best to just fix it up in a later PR, so thanks 🙂

**Comment by p-jackson** [comment]
That's a good point @vykes-mac, we could certainly misuse Emotion. If we end up needing animations or some other frequently updated styles we should probably use inline styles in those cases I think.
icymi there was a short discussion here about Emotion vs. Tailwind: https://github.com/Automattic/wp-calypso/pull/64965#discussion_r907485237

**Review by wojtekn — APPROVED** [review]

**Comment by sejas** [comment]
The tests were failing with this error:

Screenshot 2022-07-05 at 11 25 57

>   ● Test suite failed to run
>    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string
>    Consider using the "jsdom" test environment.
>    ReferenceError: window is not defined

To avoid this issue in the tests, the components that import `@wordpress/components` cannot live inside of `client/components`, because it introduces multiple references to the `window` object, which doesn't exist in our tests environment.
For this project, a good place to add these components is `client/sites-dashboard/components`.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7419382

Thank you @p-jackson for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7419382).

**Pull request merged**